jwk.go

```diff
@@ -233,15 +233,29 @@ func (k *JSONWebKey) UnmarshalJSON(data []byte) (err error) {
 
 	if certPub != nil && keyPub != nil {
 		if !reflect.DeepEqual(certPub, keyPub) {
+			fmt.Printf("%s\n", certPub)
+			fmt.Printf("%s\n", keyPub)
 			return errors.New("square/go-jose: invalid JWK, public keys in key and x5c fields to not match")
 		}
 	}
 
 	*k = JSONWebKey{Key: key, KeyID: raw.Kid, Algorithm: raw.Alg, Use: raw.Use, Certificates: certs}
 
 	k.CertificatesURL = raw.X5u
-	k.CertificateThumbprintSHA1 = raw.X5tSHA1.bytes()
-	k.CertificateThumbprintSHA256 = raw.X5tSHA256.bytes()
+
+	// x5t parameters are base64url-encoded SHA thumbprints
+	// See RFC 7517, Section 4.8, https://tools.ietf.org/html/rfc7517#section-4.8
+	x5tSHA1bytes, err := base64.RawURLEncoding.DecodeString(raw.X5tSHA1)
+	if err != nil {
+		return errors.New("square/go-jose: invalid JWK, x5t header has invalid encoding")
+	}
+	k.CertificateThumbprintSHA1 = x5tSHA1bytes
```

Comment:
For me this is still a problem as it doesn't really return 20 bytes but 40, which is always different from sha1.Sum which returns [20]byte. 

The example auth0 keys in the issue also have this problem for me.

Reply:
Which also means that line 262 always fails for me

Reply:
ok, so my understanding was that `x5t` is base64url-encoded SHA in raw bytes form, not a hex.

Reply:
So for this one I think that was the problem, it's a base64url-encoded SHA hex. Although I don't think this is standard as I understand it, it would still be good to attempt it and fallback to assuming it was raw format if it fails?

Reply:
Adding on line 254 works for me, Marshalling still has a problem afterwards where bytes.Equal is returning false for the sha1 check although the values look the same, I'll be able to look into this further tomorrow 
```
	hx, err := hex.DecodeString(string(x5tSHA1bytes))
	if err == nil {
		x5tSHA1bytes = hx
	}
```

Reply:
I'm still pondering this, but if the thumbprint was a hex then it wouldn't need to be base64-encoded.

This might be tangential (since it's not directly discussing `x5t`), but the [JSON Web Key (JWK) Thumbprint](https://tools.ietf.org/html/rfc7638) explicitly mentions base64-encoding raw bytes of the SHA output.

Reply:
To be fair the JWK RFC also only mentions Base64 Sha1 for the field specifically
```
base64url-encoded SHA-1 thumbprint
```

The problem right now is that go-auth0 uses go-jose to parse the JWKs and they use hex so although it is a bit strange to include would be great to not break it either if possible

Reply:
I guess either that or introducing an option to make this validation optional would work as a non ideal alternative